src/components/xpn-design.vue

```diff
@@ -0,0 +1,378 @@
+                return;
+            }
+
+            this.grid = JSON.parse(localStorage.getItem('xpn-data') || '[]');
+
+            if(this.grid.length != 0) {
+                return;
+            }
+
+            this.grid.push(
+                [
+                    {type:'response', value:'?'},
+                    {type:'service', label:'New Service', value:''},
+                    {type:'task', value:'?'}
+                ]
+            );
+
+            this.grid.push(
+                [
+                    {type:'obj', label:'New Object', value:''},
+                    {type:'rules', value:[
+                        {type:'rule', label:'New Rule', value:''}
+                    ]},
+                    {type:'subject', label:'New Actor', value:''}
+                ]
+            );
+
+            this.grid.push(
+                [
+                    {type:'message', value:'?'},
+                    {type:'action', label:'New Action', value:''},
+                    {type:'request', value:'?'}
+                ]
+            );
+        },
+
+        save() {
+            localStorage.setItem('xpn-data', JSON.stringify(this.grid) );
+            /// TODO: show confirmation notification
+        },
+
+        clear() {
+            for(let i = 0; i < this.grid.length; i++) {
+                this.grid.splice(i, this.grid.length);
+            }
+
+            this.init();
+        },
+
+        findMidColofMidRow() {
+            let midRowIndex = this.findMidRow();
+
+            for(let i = 0; i<this.grid[midRowIndex].length; i++) {
+                if(this.grid[midRowIndex][i].type == 'rules') {
+                    return i;
+                }
+            }
+        },
+
+        findMidColOfTopRow() {
+            for(let i = 0; i<this.grid[0].length; i++) {
+                if(this.grid[0][i].type == 'service') {
+                    return i;
+                }
+            }
+        },
+
+        findMidColOfBottomRow() {
+            let index = this.grid.length - 1;
+            for(let i = 0; i<this.grid[index].length; i++) {
+                if(this.grid[index][i].type == 'action') {
+                    return i;
+                }
+            }
+        },
+
+        findMidRow() {
+            for(let i = 0; i<this.grid.length; i++) {
+                if(this.grid[i][0].type == 'obj') {
+                    return i;
+                }
+            }
+        },
+
+        findCol(obj) {
+            for(let i = 0; i<this.grid.length; i++) {
+                for(let j = 0; j<this.grid[i].length; j++) {
+                    if(this.grid[i][j] == obj) {
+                        return j;
+                    }
+                }
+            }
+        },
+
+        findRules() {
+            let midRowIndex = this.findMidRow();
+            let midColIndex = this.findMidColofMidRow();
+
+            return this.grid[midRowIndex][midColIndex].value;
+        },
+
+        createTopRow() {
```

Comment:
We don't need these event handling methods since we are not adding/deleting any rows/cols in DESIGN mode.